pkg/serviceaccounts/oauthclient/oauthclientregistry.go

```diff
@@ -0,0 +1,88 @@
+package oauthclient
+
+import (
+	"strconv"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+
+	scopeauthorizer "github.com/openshift/origin/pkg/authorization/authorizer/scope"
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/oauthclient"
+)
+
+const UseAsOAuthClientSecretAnnotation = "serviceaccounts.openshift.io/use-as-client-secret"
```

Comment:
@openshift/api-review This annotation allows a user to opt-in to using their SA as a valid oauth client and controls the set of tokens that are allowed to used as the client secret.